lib/schema.js

```diff
@@ -305,73 +305,88 @@ Schema.prototype.paths;
 Schema.prototype.tree;
 
 /**
- * Returns a deep copy of the schema
- *
- * ####Example:
- *
- *     const schema = new Schema({ name: String });
- *     const clone = schema.clone();
- *     clone === schema; // false
- *     clone.path('name'); // SchemaString { ... }
- *
  * @return {Schema} the cloned schema
  * @api public
  * @memberOf Schema
  * @instance
  */
 
-Schema.prototype.clone = function() {
-  const s = this._clone();
+Schema.prototype.clone = function schemaDeepCopy() {
```

Comment:
Using different terminology(`schemaDeepCopy` vs `schema.clone`) can be a bit confusing in the long run.
Perhaps `function cloneSchema() {...}` would be more consistent?

Reply:
I agree, this change doesn't seem necessary or helpful. Maybe `schemaClone()`, but I'd prefer to make this `function clone()`